scripts/irods/test/test_itrim.py

```diff
@@ -218,3 +296,326 @@ def test_itrim_displays_incorrect_count__issue_3531(self):
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+            # -N cannot be 0.
+            self.admin.assert_icommand(['itrim', '-N0', logical_path], 'STDERR', 'SYS_INVALID_INPUT_PARAM')
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, self.admin.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+            # -N cannot be negative.
+            self.admin.assert_icommand(['itrim', '-N"-1"', logical_path], 'STDERR', 'SYS_INVALID_INPUT_PARAM')
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, self.admin.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+            # -N cannot overflow. Maximum value is INT32_MAX (2^31 - 1 = 2147483647). Should be enough for anyone.
+            self.admin.assert_icommand(
+                ['itrim', '-N2147483648', logical_path], 'STDERR', 'SYS_INVALID_INPUT_PARAM')
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, self.admin.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+            # -N must be a number.
+            self.admin.assert_icommand(['itrim', '-Nnope', logical_path], 'STDERR', 'SYS_INVALID_INPUT_PARAM')
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, self.admin.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+            # -N must be an Integer.
+            self.admin.assert_icommand(['itrim', '-N"3.14"', logical_path], 'STDERR', 'SYS_INVALID_INPUT_PARAM')
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, self.admin.default_resource))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc1))
+            self.assertTrue(lib.replica_exists_on_resource(self.admin, logical_path, resc2))
+
+        finally:
+            self.admin.assert_icommand(['ils', '-l', logical_path], 'STDOUT', filename) # debugging
+            self.admin.run_icommand(['irm', '-f', logical_path])
+            lib.remove_resource(self.admin, resc1)
+            lib.remove_resource(self.admin, resc2)
+
+
+class test_itrim_target_replica_selection_decision_making(unittest.TestCase):
```

Comment:
Missing issue number on class name?

Reply:
I don't know that this is tied to a particular issue... It's more just for making sure default `itrim` behavior is covered by tests. I guess I started thinking about it more in - and was inspired to write the test as a result of - #7468? Thoughts?

Reply:
I get what you're saying, but having that issue number gives us a point of reference for when we have to revisit the tests and figure out why they were introduced. _Git history also achieves this if we're committing things in a reasonable fashion._

If iRODS was in pre-1.0 status, the issue number would be less important to me. But given the history of the repo, I feel it's helpful to include the issue number.

Willing to be persuaded otherwise.

Have we added tests recently which do not have issue numbers attached to them?

Reply:
Right, I'm in agreement about the issue numbers. I just don't know that there *is* an issue number to associate here. #7468 is the closest thing I can come up with, and it's not strictly speaking related to all (if any) of these tests. Plus, I already wrote a test which directly addresses that issue in `test_itrim.Test_Itrim.test_itrim_does_not_remove_the_last_replica__issue_7468`.

Here's a recent-ish commit in which I introduced several tests with no issue number: 78c3dca6dfdc423cc8e091be35adda7a08fc1ffe

Again, I definitely think we should add issue numbers. Just looking for suggestions on an issue to link to in this case since it's not clear to me which issue would make sense here.

Reply:
I see. You could create a new issue that captures the reasons for adding the tests. You could even link other related issues to the new issue so there's a link between them.

Reply:
Created https://github.com/irods/irods/issues/7515 and included the issue in the title.